docs/source/training.rst

```diff
@@ -10,290 +10,362 @@
+more information.
 
-Fine-tuning in native PyTorch
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+First, the :obj:`load_dataset` function downloads and caches the dataset in one line of code:
+
+.. code-block:: python
 
-Model classes in 🤗 Transformers that don't begin with ``TF`` are `PyTorch Modules
-<https://pytorch.org/docs/master/generated/torch.nn.Module.html>`_, meaning that you can use them just as you would any
-model in PyTorch for both inference and optimization.
+    from datasets import load_dataset
 
-Let's consider the common task of fine-tuning a masked language model like BERT on a sequence classification dataset.
-When we instantiate a model with :func:`~transformers.PreTrainedModel.from_pretrained`, the model configuration and
-pre-trained weights of the specified model are used to initialize the model. The library also includes a number of
-task-specific final layers or 'heads' whose weights are instantiated randomly when not present in the specified
-pre-trained model. For example, instantiating a model with
-``BertForSequenceClassification.from_pretrained('bert-base-uncased', num_labels=2)`` will create a BERT model instance
-with encoder weights copied from the ``bert-base-uncased`` model and a randomly initialized sequence classification
-head on top of the encoder with an output size of 2. Models are initialized in ``eval`` mode by default. We can call
-``model.train()`` to put it in train mode.
+    raw_datasets = load_dataset("imdb")
+
+This works like the :obj:`from_pretrained` method we saw for the models and tokenizers (except the cache directory is
+`~/.cache/huggingface/dataset` by default).
+
+The :obj:`raw_datasets` objects is a dictionary with three keys: :obj:`"train"`, :obj:`"test"` and
+:obj:`"unsupervised"` (which correspond to the three splits of that dataset). We will use the :obj:`"train"` split for
+training and the :obj:`"test"` split for validation.
+
+To preprocess our data, we will need a tokenizer:
 
 .. code-block:: python
 
-    from transformers import BertForSequenceClassification
-    model = BertForSequenceClassification.from_pretrained('bert-base-uncased')
-    model.train()
+    from transformers import AutoTokenizer
+
+    tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
 
-This is useful because it allows us to make use of the pre-trained BERT encoder and easily train it on whatever
-sequence classification dataset we choose. We can use any PyTorch optimizer, but our library also provides the
-:func:`~transformers.AdamW` optimizer which implements gradient bias correction as well as weight decay.
+As we saw in :doc:`preprocessing`, we can prepare the inputs texts for the model is done with the following command:
 
 .. code-block:: python
 
-    from transformers import AdamW
-    optimizer = AdamW(model.parameters(), lr=1e-5)
+    inputs = tokenizer(sentences, padding="max_length", truncation=True)
```

Comment:
```suggestion
    sentences = raw_datasets['train']['text']
    inputs = tokenizer(sentences, padding="max_length", truncation=True)
```

Reply:
Adding this because `sentences` doesn't exist, but I think I understood what you meant!  I reworded the bit below to emphasize that this is just an example for illustrative purposes.

Reply:
This wasn't meant to be a code sample you can execute, just an example.